spacewalk/admin/uyuni-update-config

```diff
@@ -4,32 +4,52 @@
 import sys
 import os.path
 import uuid
+import subprocess
 from spacewalk.common.rhnConfig import initCFG, CFG
 
 initCFG('server.susemanager')
-try:
-    if CFG.scc_backup_srv_usr:
-        # nothing to do
-        sys.exit(0)
-except AttributeError:
-    # key does not exist, we need to create it
-    pass
-
-scc_cred_file = "/etc/zypp/credentials.d/SCCcredentials"
-
-uuidNum = None
-if os.path.exists(scc_cred_file):
-    with open(scc_cred_file, "r") as f:
-        for line in f:
-            if line.startswith("username"):
-                _k, v = line.split("=", 2)
-                uuidNum = v.strip()
-                break
-if not uuidNum:
-    # scc expects either a SCC machine login (must exists in SCC)
-    # or a UUID4 following rfc4122 to identify a anonyme proxy
-    uuidNum = str(uuid.uuid4())
-with open("/etc/rhn/rhn.conf", "a") as r:
-    r.write("\n")
-    r.write("server.susemanager.scc_backup_srv_usr = {}\n".format(uuidNum))
+
+def initSccLogin():
+
+    try:
+        if CFG.scc_backup_srv_usr:
+            # nothing to do
+            return
+    except AttributeError:
+        # key does not exist, we need to create it
+        pass
+
+    scc_cred_file = "/etc/zypp/credentials.d/SCCcredentials"
+
+    uuidNum = None
+    if os.path.exists(scc_cred_file):
+        with open(scc_cred_file, "r") as f:
+            for line in f:
+                if line.startswith("username"):
+                    _k, v = line.split("=", 2)
+                    uuidNum = v.strip()
+                    break
+    if not uuidNum:
+        # scc expects either a SCC machine login (must exists in SCC)
+        # or a UUID4 following rfc4122 to identify a anonyme proxy
+        uuidNum = str(uuid.uuid4())
+    with open("/etc/rhn/rhn.conf", "a") as r:
+        r.write("\n")
+        r.write("server.susemanager.scc_backup_srv_usr = {}\n".format(uuidNum))
+
+def importSumaGPGKeyring():
+
+    result = subprocess.run(
+            ["/usr/sbin/import-suma-build-keys"],
+            stdout=subprocess.PIPE, stderr=subprocess.STDOUT,
+            encoding='utf-8')
```

Comment:
Wouldn't it be easier to directly implement the GPG import in python rather than using a shell script that is called only here?

Reply:
GPG command line is horrible and I just copied the post script from the RPM to not introduce errors.
I know the post script is working.
Porting it to python just take time and can create more errors.
I don't want to do this.

Reply:
If you don't want to do this, then create a card to schedule this for later or we'll live with a monster for ever.

Reply:
I do not see this as a monster. For me a python port would be a monster. But ok, I will create a card for later.